<a href="https://colab.research.google.com/github/bunnybbear/BasicProg_DataViz65/blob/main/note_project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


# ราคาบ้านเดี่ยวในกรุงเทพและปริมณฑล พร้อมกับคะแนนอาหารการกินของย่านต่างๆ และคะแนนการเดินทางเข้าถึงสถานที่ต่างๆ ในชีวิตประจำวัน เพื่อประกอบการตัดสินใจซื้อบ้านเดี่ยว

## ข้อมูลที่อยู่อาศัยราย Unit

In [3]:
open_unit = pd.read_csv('/content/drive/MyDrive/Project Dataviz/Project Dataviz/opendata_unittype/opendata_unittype.csv')
open_unit.head()

row_number     unittype_id    project_id             name_th  \
0       27501   unittype_9565  project_2446           2 Bedroom   
1       28777  unittype_10267  project_3543  บ้านแฝดสองชั้น Z-2   
2       28865    unittype_471   project_143    1 Bedroom Type B   
3       24988  unittype_19506  project_7766    Studio Floor 1-2   
4       38596  unittype_10290  project_3551                TH 3   

   propertytype_id propertytype_name_en propertytype_name_th  \
0                2                Condo                คอนโด   
1            20000           Twin house              บ้านแฝด   
2                2                Condo                คอนโด   
3                2                Condo                คอนโด   
4                3             Townhome             ทาวน์โฮม   

   area_total_min_wa  area_usable_min  price_min  ...  count_room_guest  \
0                NaN             37.3  2500000.0  ...               1.0   
1               40.0            150.0  2390000.0  ...               NaN   
2                NaN             28.0        NaN  ...               NaN   
3                NaN             27.0   170000.0  ...               NaN   
4                NaN            182.0  2390000.0  ...               1.0   

   count_room_kitchen  count_room_living  count_room_maid  count_room_storage  \
0                 1.0                NaN              NaN                 NaN   
1                 NaN                NaN              NaN                 NaN   
2                 NaN                NaN              NaN                 NaN   
3                 NaN                NaN              NaN                 NaN   
4                 1.0                NaN              NaN                 2.0   

   count_room_utility         date_created         date_updated  source  \
0                 NaN  2021-11-19 18:00:09  2021-11-19 17:44:45  baania   
1                 NaN  2022-01-27 14:30:04  2022-01-27 14:22:55  baania   
2                 NaN  2022-01-14 15:00:10  2022-01-14 14:40:04  baania   
3                 NaN  2020-12-04 10:40:59  2020-10-14 01:06:48  baania   
4                 NaN  2021-11-12 13:30:08  2021-02-03 12:12:54  baania   

                                        url_unittype  
0  https://baania.com/th/project/เฟ-ลิค-คอนโด-ลาด...  
1  https://baania.com/th/project/เซน-บาย-ศิวาลัย-...  
2  https://baania.com/th/project/วันพลัส-นายน์ทีน...  
3  https://baania.com/th/project/บ้านสวนชมน้ำ-776...  
4  https://baania.com/th/project/เบล็สทาวน์-ศรีนค...  

[5 rows x 27 columns]

In [4]:
#Select columns
open_unit.columns

Index(['row_number', 'unittype_id', 'project_id', 'name_th', 'propertytype_id',
       'propertytype_name_en', 'propertytype_name_th', 'area_total_min_wa',
       'area_usable_min', 'price_min', 'count_airconditioner',
       'count_elevator', 'count_floor_internal', 'count_parking',
       'count_room_bath', 'count_room_bed', 'count_room_dinning',
       'count_room_guest', 'count_room_kitchen', 'count_room_living',
       'count_room_maid', 'count_room_storage', 'count_room_utility',
       'date_created', 'date_updated', 'source', 'url_unittype'],
      dtype='object')

In [5]:
unit = open_unit[['unittype_id', 'project_id','propertytype_id','propertytype_name_en', 'propertytype_name_th', 'area_total_min_wa','price_min']]

In [6]:
#เช็ค missing ของ unit
unit.isnull().any()

unittype_id             False
project_id              False
propertytype_id         False
propertytype_name_en    False
propertytype_name_th    False
area_total_min_wa        True
price_min                True
dtype: bool

In [7]:
unit['area_total_min_wa'] = unit['area_total_min_wa'].fillna('1')
unit['price_min'] = unit['price_min'].fillna('0')

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [8]:
unit.isnull().any() #เช็ค missing อีกครั้ง

unittype_id             False
project_id              False
propertytype_id         False
propertytype_name_en    False
propertytype_name_th    False
area_total_min_wa       False
price_min               False
dtype: bool

In [9]:
unit

unittype_id                project_id  propertytype_id  \
0                 unittype_9565              project_2446                2   
1                unittype_10267              project_3543            20000   
2                  unittype_471               project_143                2   
3                unittype_19506              project_7766                2   
4                unittype_10290              project_3551                3   
...                         ...                       ...              ...   
42620            unittype_38508             project_16864                1   
42621            unittype_14016              project_5275                1   
42622  617bde1687dd2300149d2195  617bdd47f62011001350de07                3   
42623            unittype_37375             project_16389                1   
42624            unittype_17011              project_6762                1   

      propertytype_name_en propertytype_name_th area_total_min_wa  price_min  
0                    Condo                คอนโด                 1  2500000.0  
1               Twin house              บ้านแฝด              40.0  2390000.0  
2                    Condo                คอนโด                 1          0  
3                    Condo                คอนโด                 1   170000.0  
4                 Townhome             ทาวน์โฮม                 1  2390000.0  
...                    ...                  ...               ...        ...  
42620       Detached House                 บ้าน                 1          0  
42621       Detached House                 บ้าน                 1  7490000.0  
42622             Townhome             ทาวน์โฮม                 1          0  
42623       Detached House                 บ้าน                 1  3200000.0  
42624       Detached House                 บ้าน                 1  3790000.0  

[42625 rows x 7 columns]

## ข้อมูลที่อยู่อาศัยรายโครงการ Project

In [10]:
open_project = pd.read_csv('/content/drive/MyDrive/Project Dataviz/Project Dataviz/opendata_project/opendata_project.csv')
open_project

row_number                project_id  \
0          7880.0  5ee08d64dda791001299b2a6   
1         16689.0             project_11326   
2           290.0  62739c4089233e001362b001   
3         20719.0              project_2010   
4           199.0             project_16838   
...           ...                       ...   
23429     12883.0             project_12317   
23430     21181.0             project_12057   
23431     13993.0              project_4461   
23432      5889.0               project_144   
23433      5529.0             project_16300   

                                            name_en  \
0                Winning Residence Sukhumvit Praksa   
1                                  Gift Condominium   
2                  Sanrak 9 Sukprayoon Nong Krathum   
3                                   Baan Sansabai 1   
4                        Baanpruksa Ratchapruek-345   
...                                             ...   
23429  Baan Klangmeung Urbanion Ladphrao-Chokchai 4   
23430                           @Home Kaset-Nawamin   
23431                    I-Design Wongwaen-Ramintra   
23432                 Koolpunt Ville 15 Park Avenue   
23433                          Thawisuk Garden Home   

                                             name_th propertytype_id  \
0                 วินนิ่ง เรสซิเดนท์ สุขุมวิท แพรกษา               1   
1                                กิ๊ฟท์ คอนโดมิเนียม               2   
2                     แสนรัก 9 ศุขประยูร หนองกระทุ่ม               3   
3                                      บ้านแสนสบาย 1               1   
4                             บ้านพฤกษา ราชพฤกษ์-345               3   
...                                              ...             ...   
23429  บ้านกลางเมือง เออร์บาเนี่ยน ลาดพร้าว-โชคชัย 4               3   
23430                          แอทโฮม เกษตร-นวมินทร์               3   
23431                     ไอ-ดีไซน์ วงแหวน-รามอินทรา               4   
23432                  กุลพันธ์วิลล์ 15 พาร์ค อเวนิว               1   
23433                             ทวีสุข การ์เด้นโฮม               1   

      propertytype_name_en propertytype_name_th  price_min     developer_id  \
0           Detached House                 บ้าน  3790000.0  developer_10678   
1                    Condo                คอนโด  1100000.0   developer_7319   
2                 Townhome             ทาวน์โฮม        0.0  developer_10510   
3           Detached House                 บ้าน        0.0  developer_10510   
4                 Townhome             ทาวน์โฮม  1930000.0   developer_1026   
...                    ...                  ...        ...              ...   
23429             Townhome             ทาวน์โฮม  5700000.0   developer_1760   
23430             Townhome             ทาวน์โฮม  3590000.0   developer_1793   
23431           Commercial         อาคารพาณิชย์  1870000.0   developer_2217   
23432       Detached House                 บ้าน  4000000.0   developer_1004   
23433       Detached House                 บ้าน  2800000.0  developer_10510   

                   developer_name_en  ... facility_meeting  facility_park  \
0       Winning Asset Plus Co., Ltd.  ...              NaN            NaN   
1           Gift Asset Land Co.,Ltd.  ...              NaN            1.0   
2                                NaN  ...              NaN            NaN   
3                                NaN  ...              NaN            NaN   
4       Preuksa Real Estate Co.,Ltd.  ...              NaN            1.0   
...                              ...  ...              ...            ...   
23429         AP (Thailand) Co,. Ltd  ...              NaN            1.0   
23430       Areeya Property Co.,Ltd.  ...              NaN            NaN   
23431  Wangthong Group Pub Co., Ltd.  ...              NaN            1.0   
23432     Homeland Service Co., Ltd.  ...              NaN            1.0   
23433                            NaN  ...              NaN            NaN   

       facility_playground facility_pool f

In [11]:
#Select columns
open_project.columns

Index(['row_number', 'project_id', 'name_en', 'name_th', 'propertytype_id',
       'propertytype_name_en', 'propertytype_name_th', 'price_min',
       'developer_id', 'developer_name_en', 'developer_name_th', 'latitude',
       'longitude', 'neighborhood_id', 'neighborhood_name_en',
       'neighborhood_name_th', 'subdistrict_id', 'subdistrict_name_en',
       'subdistrict_name_th', 'district_id', 'district_name_en',
       'district_name_th', 'province_id', 'province_name_en',
       'province_name_th', 'zipcode', 'count_elevator',
       'count_elevator_service', 'count_floor', 'count_parking', 'count_tower',
       'count_unit', 'count_unittype', 'facility_clubhouse',
       'facility_fitness', 'facility_meeting', 'facility_park',
       'facility_playground', 'facility_pool', 'facility_security',
       'date_created', 'date_finish', 'date_updated', 'source', 'url_project'],
      dtype='object')

In [12]:
project = open_project[['project_id','name_en','name_th','subdistrict_id','subdistrict_name_en','subdistrict_name_th','district_id','district_name_en','district_name_th','province_id','province_name_en','province_name_th']]

In [13]:
#เช็ค missing ของ project
project.isnull().any()

project_id             False
name_en                False
name_th                False
subdistrict_id          True
subdistrict_name_en     True
subdistrict_name_th     True
district_id             True
district_name_en        True
district_name_th        True
province_id             True
province_name_en        True
province_name_th        True
dtype: bool

clean subdis

In [14]:
project['subdistrict_id'].isnull().any()

True

In [15]:
project['subdistrict_id']

0        3510.0
1        4700.0
2        4672.0
3        2861.0
4        3603.0
          ...  
23429    3958.0
23430    3958.0
23431    3794.0
23432    1067.0
23433    3483.0
Name: subdistrict_id, Length: 23434, dtype: float64

In [16]:
C_subdistrict_id = project['subdistrict_id'].dropna()

In [17]:
C_subdistrict_id.isnull()

0        False
1        False
2        False
3        False
4        False
         ...  
23429    False
23430    False
23431    False
23432    False
23433    False
Name: subdistrict_id, Length: 23415, dtype: bool

ก่อน drop มี 23434 หลัง drop มี 23415 เพราะงั้นตัดได้

clean subdis name

In [18]:
project['subdistrict_name_th'].isnull().any()

True

In [19]:
project['subdistrict_name_th']

0           แพรกษา
1           บ่อวิน
2        หนองตำลึง
3             จอหอ
4         บางคูวัด
           ...    
23429     ลาดพร้าว
23430     ลาดพร้าว
23431       บางชัน
23432     สันปูเลย
23433      บ้านฉาง
Name: subdistrict_name_th, Length: 23434, dtype: object

In [20]:
C_subdistrict_name_th = project['subdistrict_name_th'].dropna()

In [21]:
C_subdistrict_name_th.isnull().any()

False

In [22]:
C_subdistrict_name_th

0           แพรกษา
1           บ่อวิน
2        หนองตำลึง
3             จอหอ
4         บางคูวัด
           ...    
23429     ลาดพร้าว
23430     ลาดพร้าว
23431       บางชัน
23432     สันปูเลย
23433      บ้านฉาง
Name: subdistrict_name_th, Length: 23418, dtype: object

ก่อน drop มี 23434 หลัง drop มี 23418 เพราะงั้นตัดได้

clean district_id 

In [23]:
project['district_id'].isnull().any()

True

In [24]:
project['district_id']

0        3499.0
1        4696.0
2        4663.0
3        2858.0
4        3600.0
          ...  
23429    3957.0
23430    3957.0
23431    3791.0
23432    1054.0
23433    3482.0
Name: district_id, Length: 23434, dtype: float64

In [25]:
C_district_id = project['district_id'].dropna()

In [26]:
C_district_id

0        3499.0
1        4696.0
2        4663.0
3        2858.0
4        3600.0
          ...  
23429    3957.0
23430    3957.0
23431    3791.0
23432    1054.0
23433    3482.0
Name: district_id, Length: 23424, dtype: float64

ก่อน drop มี 23434 หลัง drop มี 23424 เพราะงั้นตัดได้

In [27]:
project['district_name_th'].isnull().any()

True

In [28]:
project['district_name_th']

0        เมืองสมุทรปราการ
1                 ศรีราชา
2                  พานทอง
3         เมืองนครราชสีมา
4           เมืองปทุมธานี
               ...       
23429            ลาดพร้าว
23430            ลาดพร้าว
23431           คลองสามวา
23432           ดอยสะเก็ด
23433             บ้านฉาง
Name: district_name_th, Length: 23434, dtype: object

In [29]:
project['district_id'] = project['district_id'].fillna
project['province_id'] = project['province_id'].fillna('0')
project['subdistrict_id'] = project['subdistrict_id'].fillna('0')

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documen

In [30]:
project

project_id                                       name_en  \
0      5ee08d64dda791001299b2a6            Winning Residence Sukhumvit Praksa   
1                 project_11326                              Gift Condominium   
2      62739c4089233e001362b001              Sanrak 9 Sukprayoon Nong Krathum   
3                  project_2010                               Baan Sansabai 1   
4                 project_16838                    Baanpruksa Ratchapruek-345   
...                         ...                                           ...   
23429             project_12317  Baan Klangmeung Urbanion Ladphrao-Chokchai 4   
23430             project_12057                           @Home Kaset-Nawamin   
23431              project_4461                    I-Design Wongwaen-Ramintra   
23432               project_144                 Koolpunt Ville 15 Park Avenue   
23433             project_16300                          Thawisuk Garden Home   

                                             name_th subdistrict_id  \
0                 วินนิ่ง เรสซิเดนท์ สุขุมวิท แพรกษา         3510.0   
1                                กิ๊ฟท์ คอนโดมิเนียม         4700.0   
2                     แสนรัก 9 ศุขประยูร หนองกระทุ่ม         4672.0   
3                                      บ้านแสนสบาย 1         2861.0   
4                             บ้านพฤกษา ราชพฤกษ์-345         3603.0   
...                                              ...            ...   
23429  บ้านกลางเมือง เออร์บาเนี่ยน ลาดพร้าว-โชคชัย 4         3958.0   
23430                          แอทโฮม เกษตร-นวมินทร์         3958.0   
23431                     ไอ-ดีไซน์ วงแหวน-รามอินทรา         3794.0   
23432                  กุลพันธ์วิลล์ 15 พาร์ค อเวนิว         1067.0   
23433                             ทวีสุข การ์เด้นโฮม         3483.0   

      subdistrict_name_en subdistrict_name_th  \
0                PHRAEKSA              แพรกษา   
1                  BO WIN              บ่อวิน   
2           NONG TAMLUENG           หนองตำลึง   
3                  CHO HO                จอหอ   
4            BANG KHU WAT            บางคูวัด   
...                   ...                 ...   
23429           LAT PHRAO            ลาดพร้าว   
23430           LAT PHRAO            ลาดพร้าว   
23431           BANG CHAN              บางชัน   
23432         SAN PU LOEI            สันปูเลย   
23433           BAN CHANG             บ้านฉาง   

                                             district_id  \
0      <bound method Series.fillna of 0        3499.0...   
1      <bound method Series.fillna of 0        3499.0...   
2      <bound method Series.fillna of 0        3499.0...   
3      <bound method Series.fillna of 0        3499.0...   
4      <bound method Series.fillna of 0        3499.0...   
...                                                  ...   
23429  <bound method Series.fillna of 0        3499.0...   
23430  <bound method Series.fillna of 0        3499.0...   
23431  <bound method Series.fillna of 0        3499.0...   
23432  <bound method Series.fillna of 0        3499.0...   
23433  <bound method Series.fillna of 0        3499.0...   

               district_name_en  district_name_th province_id  \
0           MUEANG SAMUT PRAKAN  เมืองสมุทรปราการ      3498.0   
1                      SI RACHA           ศรีราชา      4616.0   
2                    PHAN THONG            พานทอง      4616.0   
3      MUEANG NAKHON RATCHASIMA   เมืองนครราชสีมา      2857.0   
4           MUEANG PATHUM THANI     เมืองปทุมธานี      3599.0   
...                         ...               ...         ...   
23429                 LAT PHRAO          ลาดพร้าว      3781.0   
23430                 LAT PHRAO          ลาดพร้าว      3781.0   
23431             KHLONG SAM WA         คลองสามวา      3781.0   
23432                 DOI SAKET         ดอยสะเก็ด      1005.0   
23433                 BAN CHANG           บ้านฉาง      3431.0   

        province_name_en province_name_th  
0           Samut Prakan      สมุทรปราการ  
1              C

## คะแนนเกี่ยวกับอาหารการกินของย่าน (Subdistrict) ต่างๆ

In [31]:
open_eat = pd.read_csv('/content/drive/MyDrive/Project Dataviz/Project Dataviz/opendata_eating_score/opendata_eating_score.csv')
open_eat.head()

row_number  subdistrict_id subdistrict_name_en subdistrict_name_th  \
0         145            1007         CHANG KHLAN            ช้างคลาน   
1         102            3941     BANG PHONGPHANG           บางโพงพาง   
2          64            3829            CHIMPHLI              ฉิมพลี   
3          96            3503           BANG DUAN             บางด้วน   
4          17            2624               RAWAI              ราไวย์   

   district_id     district_name_en  district_name_th  province_id  \
0         1006    MUEANG CHIANG MAI    เมืองเชียงใหม่         1005   
1         3939             YAN NAWA           ยานนาวา         3781   
2         3826          TALING CHAN          ตลิ่งชัน         3781   
3         3499  MUEANG SAMUT PRAKAN  เมืองสมุทรปราการ         3498   
4         2617        MUEANG PHUKET       เมืองภูเก็ต         2616   

  province_name_en province_name_th  eating_daytime  eating_daytime_price  \
0       Chiang Mai        เชียงใหม่            64.0                  48.0   
1          Bangkok    กรุงเทพมหานคร            71.0                  73.0   
2          Bangkok    กรุงเทพมหานคร            73.0                  84.0   
3     Samut Prakan      สมุทรปราการ            79.0                  93.0   
4           Phuket           ภูเก็ต            52.0                  73.0   

   eating_daytime_quality  eating_daytime_variety  eating_nighttime  \
0                    62.0                    97.0              73.0   
1                    64.0                    82.0              73.0   
2                    65.0                    72.0              70.0   
3                    93.0                    21.0              62.0   
4                    36.0                    52.0              48.0   

   eating_nighttime_price  eating_nighttime_quality  eating_nighttime_variety  \
0                    41.0                      70.0                      92.0   
1                    62.0                      65.0                      85.0   
2                    77.0                      60.0                      73.0   
3                    76.0                      69.0                      49.0   
4                    67.0                      34.0                      48.0   

   source         day  
0  baania  2022-09-30  
1  baania  2022-09-30  
2  baania  2022-09-30  
3  baania  2022-09-30  
4  baania  2022-09-30

In [32]:
#Select columns
open_eat.columns

Index(['row_number', 'subdistrict_id', 'subdistrict_name_en',
       'subdistrict_name_th', 'district_id', 'district_name_en',
       'district_name_th', 'province_id', 'province_name_en',
       'province_name_th', 'eating_daytime', 'eating_daytime_price',
       'eating_daytime_quality', 'eating_daytime_variety', 'eating_nighttime',
       'eating_nighttime_price', 'eating_nighttime_quality',
       'eating_nighttime_variety', 'source', 'day'],
      dtype='object')

In [33]:
eat = open_eat[['district_id','province_id','subdistrict_id', 'subdistrict_name_en',
       'subdistrict_name_th', 'eating_daytime', 'eating_daytime_price',
       'eating_daytime_quality', 'eating_daytime_variety', 'eating_nighttime',
       'eating_nighttime_price', 'eating_nighttime_quality',
       'eating_nighttime_variety']]

In [34]:
#เช็ค missing ของ eat
eat.isnull().any()

district_id                 False
province_id                 False
subdistrict_id              False
subdistrict_name_en         False
subdistrict_name_th         False
eating_daytime              False
eating_daytime_price        False
eating_daytime_quality      False
eating_daytime_variety      False
eating_nighttime            False
eating_nighttime_price      False
eating_nighttime_quality    False
eating_nighttime_variety    False
dtype: bool

In [35]:
eat

district_id  province_id  subdistrict_id subdistrict_name_en  \
0           1006         1005            1007         CHANG KHLAN   
1           3939         3781            3941     BANG PHONGPHANG   
2           3826         3781            3829            CHIMPHLI   
3           3499         3498            3503           BANG DUAN   
4           2617         2616            2624               RAWAI   
..           ...          ...             ...                 ...   
550         1200         1005            1211         NONG PHUENG   
551         3744         3667            3754          BANG PHASI   
552         3905         3781            3907      KHLONG MAHANAK   
553         4705         4616            4707       NA CHOM THIAN   
554         4645         4616            4648         TAKHIAN TIA   

    subdistrict_name_th  eating_daytime  eating_daytime_price  \
0              ช้างคลาน            64.0                  48.0   
1             บางโพงพาง            71.0                  73.0   
2                ฉิมพลี            73.0                  84.0   
3               บางด้วน            79.0                  93.0   
4                ราไวย์            52.0                  73.0   
..                  ...             ...                   ...   
550            หนองผึ้ง            46.0                  35.0   
551             บางภาษี            28.0                  29.0   
552          คลองมหานาค            72.0                  70.0   
553          นาจอมเทียน            60.0                  49.0   
554        ตะเคียนเตี้ย            40.0                  46.0   

     eating_daytime_quality  eating_daytime_variety  eating_nighttime  \
0                      62.0                    97.0              73.0   
1                      64.0                    82.0              73.0   
2                      65.0                    72.0              70.0   
3                      93.0                    21.0              62.0   
4                      36.0                    52.0              48.0   
..                      ...                     ...               ...   
550                    48.0                    58.0              54.0   
551                    32.0                    15.0              28.0   
552                    69.0                    82.0              74.0   
553                    57.0                    83.0              62.0   
554                    36.0                    38.0              37.0   

     eating_nighttime_price  eating_nighttime_quality  \
0                      41.0                      70.0   
1                      62.0                      65.0   
2                      77.0                      60.0   
3                      76.0                      69.0   
4                      67.0                      34.0   
..                      ...                       ...   
550                    16.0                      66.0   
551                    41.0                      39.0   
552                    58.0                      69.0   
553                    26.0                      57.0   
554                    22.0                      37.0   

     eating_nighttime_variety  
0                        92.0  
1                        85.0  
2                        73.0  
3                        49.0  
4                        48.0  
..                        ...  
550                      68.0  
551                      14.0  
552                      86.0  
553                      85.0  
554                      45.0  

[555 rows x 13 columns]

##คะแนนการเดินทางเข้าถึงสถานที่ ในชีวิตประจำวันของพื้นที่ต่างๆ โดยคำนวณจาก 3 รูปแบบการเดินทาง และ 5 ประเภทของสถานที่เป้าหมายเพื่อการดำเนินชีวิตในแต่ละวัน
### คะแนนการเดินทางเข้าถึงสถานที่ คือ ความสะดวกต่างๆ เช่น ทางเดินเท้าดี หลอดไฟเยอะ สว่าง เป็นต้น
### 3 รูปแบบการเดินทาง : เดิน, ใช้ขนส่งสาธารณะ และรถส่วนตัว### 
### คะแนนสถานที่ คือ ความสะดวก ความใกล้ที่ต่างๆ เช่น อยู่ใกล้ รร ที่ทำงานเป็นต้น
### 5 ประเภทของสถานที่ : พักผ่อนหย่อนใจ, สถานที่บริการสาธารณะ, สถานศึกษา, shopping และแหล่งทำงาน

In [36]:
open_living = pd.read_csv('/content/drive/MyDrive/Project Dataviz/Project Dataviz/opendata_living_score/opendata_living_score.csv')
open_living.head()

row_number  subdistrict_id subdistrict_name_en subdistrict_name_th  \
0         634            3410            SANO LOI              โสนลอย   
1         619            3976           TALAT NOI            ตลาดน้อย   
2         284            4669             BAN KAO            บ้านเก่า   
3         209            2647            BAN THUM            บ้านทุ่ม   
4         615            3799          RAM INTHRA           รามอินทรา   

   district_id  district_name_en district_name_th  province_id  \
0         3402    BANG BUA THONG        บางบัวทอง         3372   
1         3973    SAMPHANTHAWONG      สัมพันธวงศ์         3781   
2         4663        PHAN THONG           พานทอง         4616   
3         2638  MUEANG KHON KAEN     เมืองขอนแก่น         2637   
4         3797       KHAN NA YAO         คันนายาว         3781   

  province_name_en province_name_th  ...  access_train  access_total  \
0       Nonthaburi          นนทบุรี  ...           1.0          79.0   
1          Bangkok    กรุงเทพมหานคร  ...         100.0         100.0   
2        Chon Buri           ชลบุรี  ...           0.0           0.0   
3        Khon Kaen          ขอนแก่น  ...           0.0           0.0   
4          Bangkok    กรุงเทพมหานคร  ...           5.0          39.0   

   drive_leisure  drive_public_service  drive_school  drive_shopping  \
0          100.0                 100.0         100.0           100.0   
1          100.0                  87.0         100.0           100.0   
2           79.0                  91.0          86.0            84.0   
3           73.0                  65.0          73.0            57.0   
4           99.0                  61.0          99.0            90.0   

   drive_work  drive_total  source         day  
0       100.0        100.0  baania  2022-09-30  
1       100.0         99.0  baania  2022-09-30  
2        93.0         85.0  baania  2022-09-30  
3        47.0         62.0  baania  2022-09-30  
4        84.0         89.0  baania  2022-09-30  

[5 rows x 28 columns]

In [37]:
#Select columns
open_living.columns

Index(['row_number', 'subdistrict_id', 'subdistrict_name_en',
       'subdistrict_name_th', 'district_id', 'district_name_en',
       'district_name_th', 'province_id', 'province_name_en',
       'province_name_th', 'walk_leisure', 'walk_public_service',
       'walk_school', 'walk_shopping', 'walk_work', 'walk_total',
       'access_boat', 'access_bus', 'access_train', 'access_total',
       'drive_leisure', 'drive_public_service', 'drive_school',
       'drive_shopping', 'drive_work', 'drive_total', 'source', 'day'],
      dtype='object')

In [38]:
living = open_living[['district_id','province_id','subdistrict_id', 'walk_leisure', 'walk_public_service',
       'walk_school', 'walk_shopping', 'walk_work', 'walk_total',
       'access_boat', 'access_bus', 'access_train', 'access_total',
       'drive_leisure', 'drive_public_service', 'drive_school',
       'drive_shopping', 'drive_work', 'drive_total']]

In [39]:
#เช็ค missing ของ living
living.isnull().any()

district_id             False
province_id             False
subdistrict_id          False
walk_leisure            False
walk_public_service     False
walk_school             False
walk_shopping           False
walk_work               False
walk_total              False
access_boat             False
access_bus              False
access_train            False
access_total            False
drive_leisure           False
drive_public_service    False
drive_school            False
drive_shopping          False
drive_work              False
drive_total             False
dtype: bool

In [40]:
living

district_id  province_id  subdistrict_id  walk_leisure  \
0           3402         3372            3410          96.0   
1           3973         3781            3976         100.0   
2           4663         4616            4669           7.0   
3           2638         2637            2647          11.0   
4           3797         3781            3799          89.0   
..           ...          ...             ...           ...   
704         3533         3498            3545         100.0   
705         2626         2616            2627          50.0   
706         2638         2637            2642          11.0   
707         1006         1005            1015           4.0   
708         3668         3667            3686          83.0   

     walk_public_service  walk_school  walk_shopping  walk_work  walk_total  \
0                  100.0         98.0          100.0      100.0        99.0   
1                  100.0        100.0          100.0      100.0       100.0   
2                   18.0         21.0           24.0       19.0        19.0   
3                    9.0         13.0           25.0       13.0        18.0   
4                   94.0         89.0           97.0       99.0        95.0   
..                   ...          ...            ...        ...         ...   
704                 51.0        100.0          100.0      100.0        95.0   
705                 26.0         17.0           44.0       28.0        39.0   
706                 10.0          7.0           29.0       15.0        20.0   
707                  9.0         21.0            6.0       35.0        10.0   
708                 13.0         78.0           99.0       95.0        85.0   

     access_boat  access_bus  access_train  access_total  drive_leisure  \
0           98.0       100.0           1.0          79.0          100.0   
1          100.0       100.0         100.0         100.0          100.0   
2            0.0         0.0           0.0           0.0           79.0   
3            0.0         0.0           0.0           0.0           73.0   
4            1.0        94.0           5.0          39.0           99.0   
..           ...         ...           ...           ...            ...   
704          1.0        99.0           0.0          40.0          100.0   
705          0.0         0.0           0.0           0.0           67.0   
706          0.0         0.0           0.0           0.0           82.0   
707          0.0         0.0           0.0           0.0           61.0   
708          0.0         1.0           0.0           0.0          100.0   

     drive_public_service  drive_school  drive_shopping  drive_work  \
0                   100.0         100.0           100.0       100.0   
1                    87.0         100.0           100.0       100.0   
2                    91.0          86.0            84.0        93.0   
3                    65.0          73.0            57.0        47.0   
4                    61.0          99.0            90.0        84.0   
..                    ...           ...             ...         ...   
704                  96.0         100.0           100.0       100.0   
705                  65.0          60.0            65.0        65.0   
706                  76.0          75.0            74.0        61.0   
707                  81.0          88.0            61.0        94.0   
708                  97.0          98.0           100.0       100.0   

     drive_total  
0          100.0  
1           99.0  
2           85.0  
3           62.0  
4           89.0  
..           ...  
704        100.0  
705         65.0  
706         75.0  
707         69.0  
708         99.0  

[709 rows x 19 columns]

## Merge ตารางเข้าด้วยกัน

In [41]:
unit_project = unit.merge(project, left_on='project_id', right_on='project_id')
unit_project

unittype_id                project_id  propertytype_id  \
0                 unittype_9565              project_2446                2   
1                unittype_18502              project_2446                2   
2                 unittype_9564              project_2446                2   
3                unittype_10267              project_3543            20000   
4                unittype_10267              project_3543            20000   
...                         ...                       ...              ...   
53724            unittype_21700              project_8631                4   
53725            unittype_32960             project_14046                1   
53726  617fc61e551dd10015a6933e  617fc4d3f62011001350de55                3   
53727             unittype_1459               project_711                3   
53728            unittype_37375             project_16389                1   

      propertytype_name_en propertytype_name_th area_total_min_wa  price_min  \
0                    Condo                คอนโด                 1  2500000.0   
1                    Condo                คอนโด                 1  1390000.0   
2                    Condo                คอนโด                 1  1990000.0   
3               Twin house              บ้านแฝด              40.0  2390000.0   
4               Twin house              บ้านแฝด              40.0  2390000.0   
...                    ...                  ...               ...        ...   
53724           Commercial         อาคารพาณิชย์                 1  5990000.0   
53725       Detached House                 บ้าน                 1          0   
53726             Townhome             ทาวน์โฮม                 1  4880000.0   
53727             Townhome             ทาวน์โฮม                 1          0   
53728       Detached House                 บ้าน                 1  3200000.0   

                                name_en                          name_th  \
0              Felic Ladprao-Wanghin 79  เฟ-ลิค คอนโด ลาดพร้าว-วังหิน 79   
1              Felic Ladprao-Wanghin 79  เฟ-ลิค คอนโด ลาดพร้าว-วังหิน 79   
2              Felic Ladprao-Wanghin 79  เฟ-ลิค คอนโด ลาดพร้าว-วังหิน 79   
3                        ZEN by Sivalai                  เซน บาย ศิวาลัย   
4                        ZEN by Sivalai                  เซน บาย ศิวาลัย   
...                                 ...                              ...   
53724                  Ploypailin Place                   พลอยไพลิน เพลส   
53725          The Sunrise Ocean Villas    เดอะ ซันไรซ์ โอเชียน วิลล่าส์   
53726  Baan Klang Muang Sukhumvit-Onnut   บ้านกลางเมือง สุขุมวิท-อ่อนนุช   
53727                     Baan Sangrung                      บ้านแสงรุ่ง   
53728                NarongKit Ramintra               ณรงค์กิจ รามอินทรา   

      subdistrict_id  subdistrict_name_en subdistrict_name_th  \
0             3803.0       CHANTHARAKASEM           จันทรเกษม   
1             3803.0       CHANTHARAKASEM           จันทรเกษม   
2             3803.0       CHANTHARAKASEM           จันทรเกษม   
3             1172.0        SAN KAMPHAENG            สันกำแพง   
4             1172.0        SAN KAMPHAENG            สันกำแพง   
...              ...                  ...                 ...   
53724         3953.0  KHLONG SONG TON NUN      คลองสองต้นนุ่น   
53725         2633.0             PA KHLOK             ป่าคลอก   
53726         3902.0               PRAWET              ประเวศ   
53727         1016.0              WAT KET              วัดเกต   
53728         3864.0            ANUSAWARI          อนุสาวรีย์   

                                             district_id   district_name_en  \
0      <bound method Series.fillna of 0        3499.0...          CHATUCHAK   
1      <bound method Series.fillna of 0        3499.0...          CHATUCHAK   
2      <bound method Series.fillna of 0        3499.0...          CHATUCHAK   
3      <bound method Series.fillna of 0        3499.0...      SAN KAMPHAENG   
4      <bound method Series.filln

##Visualization

In [46]:
#drop ค่าซ้ำ
#x = Eating.drop_duplicates()
#x

In [45]:
# .reset_index() เพื่อเอาตารางไปใช้